### PR TITLE
updated PAN and added GSTIN

### PR DESCRIPTION
**Pull request by akshayaponnuvel**

### CODE REVIEW

File: NumberValidator/Validators/IN/PAN.cs [hunk 1]
```diff
@@ -20,22 +19,25 @@ public bool IsValid(string pan)
...
-            if (pan.Substring(4, 4) == "0000") // Change index from 5 to 4
+
+            if (pan.Substring(5, 4) == "0000")
             {
                 throw new InvalidComponentException();
             }
-            Regex panRegex = new Regex(panPattern);
-            if (!panRegex.IsMatch(pan))
+
+            if (!Regex.IsMatch(pan, @"^[A-Z]{5}[0-9]{4}[A-Z]$"))
```
Comment (pradeepgururani) [inline NumberValidator/Validators/IN/PAN.cs]: updating regex to a new and better regexe is fine but keep the old pattern. that was more clean and good variable names were explaining what is happening.

File: NumberValidator/Validators/IN/GSTIN.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            {
+                throw new InvalidFormatException();
+            }
+
+            if (!Regex.IsMatch(gst, @"^[0-9]{2}[A-Z]{5}[0-9]{4}[A-Z][0-9A-Z]{3}$"))
+            {
+                throw new InvalidFormatException();
+            }
+
+            string panNumber = gst.Substring(2, 10);
```
Comment (pradeepgururani) [inline NumberValidator/Validators/IN/GSTIN.cs:60]: this doesn't look right. PAN number is 10 letter alpha numeric number. here you are taking 8 letters from GST and validating that as a PAN. please explain.

File: NumberValidator/Validators/IN/PAN.cs [hunk 3]
```diff
@@ -1,10 +1,9 @@
 ﻿using System;
 using System.Globalization;
 using NumberValidator.Helpers;
-using NumberValidator.Validators.AD;
 using System.Text.RegularExpressions;
 
-namespace NumberValidator.Validators.PAN
+namespace NumberValidator.Validators.IN
```
Comment (pradeepgururani) [inline NumberValidator/Validators/IN/PAN.cs:6]: renaming the namespace is good step.

File: NumberValidator/Helpers/ExtensionMethods.cs [hunk 4]
```diff
@@ -22,5 +22,17 @@ public static bool IsDigits(this string input)
...
+        public static string RemoveSpace(this  string input)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                return string.Empty;
+            }
+
+            var cleanedInput = Regex.Replace(input, @"[\s-]", "");
+
+            return cleanedInput.ToUpper().Trim();
```
Comment (pradeepgururani) [inline NumberValidator/Helpers/ExtensionMethods.cs]: there is a coding principle which is called `Single Responsibility Principle`. In a method which is saying that it `RemoveSpace`, you are also converting it ToUpper and Trimming which is not good. leave those aspects of the logic to other places in the code. Make this method handle only `RemoveSpace` part.

**Review by pradeepgururani — CHANGES_REQUESTED** [review]

File: NumberValidator/Validators/IN/GSTIN.cs (hunk 2 above)
Comment (akshayaponnuvel) [reply NumberValidator/Validators/IN/GSTIN.cs:60]: it is actually from 2nd index taking the next 10 letters

File: NumberValidator/Helpers/ExtensionMethods.cs (hunk 4 above)
Comment (akshayaponnuvel) [reply NumberValidator/Helpers/ExtensionMethods.cs]: sure , i will change it.

File: NumberValidator/Validators/IN/PAN.cs (hunk 3 above)
Comment (akshayaponnuvel) [reply NumberValidator/Validators/IN/PAN.cs:6]: thank you

File: NumberValidator/Validators/IN/PAN.cs (hunk 1 above)
Comment (akshayaponnuvel) [reply NumberValidator/Validators/IN/PAN.cs]: ok ,  i will stick with that

Comment (akshayaponnuvel) [reply NumberValidator/Validators/IN/PAN.cs]: changes done

File: NumberValidator/Helpers/ExtensionMethods.cs (hunk 4 above)
Comment (akshayaponnuvel) [reply NumberValidator/Helpers/ExtensionMethods.cs]: changes done

**Review by pradeepgururani — APPROVED** [review]

**Pull request merged**